packages/eas-cli/src/commands/worker/deploy.ts

```diff
@@ -0,0 +1,195 @@
+  static override flags = {
+    // TODO(@kitten): Allow deployment identifier to be specified
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.ProjectDir,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    Log.warn('EAS Worker Deployments are in beta and subject to breaking changes.');
+
+    const {
+      privateProjectConfig: { projectId, exp },
+      loggedIn: { graphqlClient },
+      projectDir,
+    } = await this.getContextAsync(WorkerDeploy, {
+      nonInteractive: true,
+    });
+
+    const distPath = path.resolve(projectDir, 'dist');
+    const distClientPath = path.resolve(distPath, 'client');
+    const distServerPath = path.resolve(distPath, 'server');
+    if (!(await isDirectory(distPath))) {
+      throw new Error(
+        `No "dist/" folder found at ${distPath}. Prepare your project for deployment with "npx expo export"`
+      );
+    } else if (!(await isDirectory(distClientPath))) {
+      throw new Error(
+        `No "dist/client/" folder found at ${distClientPath}. Ensure the app.json key "expo.web.output" is set to "server"`
+      );
+    } else if (!(await isDirectory(distServerPath))) {
+      throw new Error(
+        `No "dist/server/" folder found in ${distServerPath}. Ensure the app.json key "expo.web.output" is set to "server"`
+      );
+    }
+
+    async function* emitWorkerTarballAsync(
+      assetMap: WorkerAssets.AssetMap
+    ): AsyncGenerator<WorkerAssets.FileEntry> {
+      yield ['assets.json', JSON.stringify(assetMap)];
+
+      // TODO: Create manifest from user configuration
+      const manifest = { env: {} };
+      yield ['manifest.json', JSON.stringify(manifest)];
+
+      const workerFiles = WorkerAssets.listWorkerFiles(distServerPath);
+      for await (const workerFile of workerFiles) {
+        yield [`server/${workerFile.normalizedPath}`, workerFile.data];
+      }
+    }
+
+    async function uploadTarballAsync(tarPath: string): Promise<any> {
+      const uploadUrl = await getSignedDeploymentUrlAsync(graphqlClient, exp, {
+        appId: projectId,
+      });
+
+      const { response } = await uploadAsync({
+        url: uploadUrl,
+        filePath: tarPath,
+        compress: false,
+        headers: {
+          accept: 'application/json',
+        },
+      });
+      if (response.status === 413) {
+        throw new Error(
+          'Upload failed! (Payload too large)\n' +
+            `The gzipped files in "dist/server/" (at: ${distServerPath}) exceed the maximum file size.`
+        );
+      } else if (!response.ok) {
+        throw new Error(`Upload failed! (${response.statusText})`);
+      } else {
+        const json = await response.json();
+        if (!json.success || !json.result || typeof json.result !== 'object') {
+          throw new Error(json.message ? `Upload failed: ${json.message}` : 'Upload failed!');
+        }
+        return json.result;
+      }
```

Comment:
Also here, what can the user do to unblock themselves from these errors?

Reply:
So, ideally, we'll have to test which errors can occur. The problem is that Cloudflare's Upload API has several failure modes, some of which return no details at all.
Since we haven't run into any of these undocumented, non-error-coded errors, it's hard to predict when they'd occur 🤦 

Some of them can occur if JS is invalid, I think, but we'll need to test this by hand and take notes of what errors messages we get, since there's no documented list of error codes

Reply:
Made a ticket for this: https://linear.app/expo/issue/ENG-13382/test-invalid-worker-uploads-and-take-notes-of-non-error-codes-cf-api